packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -80,16 +80,49 @@ export const useStoreSwitcher = props => {
         // Change store view code and currency to be used in Apollo link request headers
         storeCode => {
             // Do nothing when store view is not present in available stores
-            if (!availableStores.get(storeCode)) return;
+            if (!availableStores.has(storeCode)) return;
 
             storage.setItem('store_view_code', storeCode);
             storage.setItem(
                 'store_view_currency',
                 availableStores.get(storeCode).currency
             );
 
-            // Refresh the page to re-trigger the queries once code/currency are saved in local storage.
-            history.go(0);
+            // Handle updating the URL if the store code should be present
+            if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+                const pathName = window.location.pathname;
+
+                // Check to see if we're on a page outside of the homepage
+                if (pathName !== '' && pathName !== '/') {
+                    const pathNameSplit = pathName.split('/');
+                    const pathStoreCode = pathNameSplit[1];
+
+                    // Is the current store code in the URL? If so replace it with the new one
+                    if (
+                        availableStores.has(pathStoreCode) &&
+                        availableStores.get(pathStoreCode).isCurrent
+                    ) {
+                        window.location.assign(
```

Comment:
Generally, `react-router` (and its `history` dependency) are our abstraction for interacting with the location. This helps with testing and will eventually be important for SSR. I think we should be using [its API](https://reactrouter.com/core/api/history) rather than `window.location` directly.

Rather than assigning to `window.location`, can we `history.push()` or `history.replace()` as necessary?

Reply:
I believe this was used to work around the basename being updated in `BrowserRouter`. If your current basename is `/en`, `history` will append after the basename but not replace it, even if you use `replace`.

If there is a way to re-assign basename from within the `BrowserRouter` we would need to do that here.